lib/src/rules/invariant_booleans.dart

```diff
@@ -100,49 +100,53 @@ void nestedOk5() {
 Set<Expression> _findConditionsCausingReturns(
         BinaryExpression node, Iterable<AstNode> nodesInDFS) =>
     nodesInDFS
+        .where(_isAnalyzedNode)
+        .where(_isConditionalStatementWithReturn(nodesInDFS))
         .takeWhile((n) => n != node.parent)
-        .where(_isIfStatementWithReturn(nodesInDFS))
         .where((_noFurtherAssignmentInvalidatingCondition(node, nodesInDFS)))
         .fold(<Expression>[], (List<Expression> previous, AstNode statement) {
-      if (statement is IfStatement) {
-        previous.add(statement.condition);
+      if (_isAnalyzedNode(statement)) {
```

Comment:
Do we need this condition? Isn't the condition already known to be true because of the `where(_isAnalyzedNode)`?